rocky/rocky/scheduler.py

```diff
@@ -189,21 +189,35 @@ class TaskNotFoundError(SchedulerError):
     message = _("Task could not be found.")
 
 
-class SchedulerClient:
+class SchedulerSession(requests.Session):
     def __init__(self, base_uri: str):
-        self.session = requests.Session()
+        super().__init__()
+        self.session = self
         self._base_uri = base_uri
-        self.is_alive()
 
+        try:
+            self.session.get(f"{self._base_uri}/health")
+        except (
+            requests.ConnectionError,
+            requests.RequestException,
+            requests.TooManyRedirects,
+            requests.ConnectTimeout,
+            requests.ReadTimeout,
+            requests.Timeout,
+        ) as error:
```

Comment:
Since these are all subclassed from `RequestException`, you might as well use just that. Unless other exceptions are coming from `RequestException` that you don't care about?

Reply:
Yes, this is a specific subset of issues.

Reply:
> Yes, this is a specific subset of issues.

It seems a bit unusual to include `requests.RequestException` in this specific list. Doing so may make the other exceptions in this list redundant. On the other hand, if you'd remove `requests.RequestException` from the exception list, does that mean we don't care about the other exceptions such as SSL errors, JSON errors, proxy errors, etc.?

Reply:
I have checked indeed that `requests.RequestException` incorporates all the other errors, or we remove `requests.RequestException` and keep the listed ones, or we just use `requests.RequestException` and it catches the list + others mentioned.